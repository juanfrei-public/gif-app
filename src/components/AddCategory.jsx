import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setIputValue] = useState("");

  const onInputChange = ({ target }) => {
    const value = target.value;
    setIputValue(value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    onNewCategory(inputValue.trim());
    if (inputValue.length <= 1) return;
    setIputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
