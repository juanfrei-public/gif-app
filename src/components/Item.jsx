import PropTypes from "prop-types";

export const Item = ({ title, url }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

Item.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
