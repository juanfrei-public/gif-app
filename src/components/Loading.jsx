export const Loading = ({ isLoading }) => isLoading && <div> Cargando . . . </div>;
