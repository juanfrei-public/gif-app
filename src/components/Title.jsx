export const Title = ({ titulo }) => <h1>{titulo}</h1>;
