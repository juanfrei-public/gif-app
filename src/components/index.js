export * from './AddCategory';
export * from './Item';
export * from './ListarItems';
export * from './Title';
export * from './Loading';