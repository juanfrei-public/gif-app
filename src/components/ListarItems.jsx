import { Item } from "./Item";
import PropTypes from "prop-types";

import { Loading, Title } from "../components";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const ListarItems = ({ items }) => {
  const { images, isLoading } = useFetchGifs(items);

  // se pasa a un hook (useFetchGifs)
  // const [images, setImages] = useState([]);
  // const getImages = async () => {
  //   const newImages = await getGifs(items);
  //   setImages(newImages);
  // };
  // useEffect(() => {
  //   getImages();
  // }, []);

  return (
    <>
      <Title titulo={items} />
      <Loading isLoading={isLoading} />
      <div className="card-grid">
        {images.map((image) => (
          <Item key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

ListarItems.propTypes = {
  items: PropTypes.string.isRequired,
};
