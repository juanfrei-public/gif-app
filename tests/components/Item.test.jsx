/** @jest-environment jsdom */
import { render, screen } from "@testing-library/react";
import { Item } from "../../src/components/Item";

describe("Pruebas sobre Item", () => {
  const titleMock = "Un titulo mock";
  const urlMock = "https://unmock.zaraza.com/";

  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<Item title={titleMock} url={urlMock} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el url y el alt indicado", () => {
    render(<Item title={titleMock} url={urlMock} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(urlMock);
    expect(alt).toBe(titleMock);
  });

  test("Debe mostrar el titulo", () => {
    render(<Item title={titleMock} url={urlMock} />);
    expect(screen.getByText(titleMock)).toBeTruthy();
  });
});
