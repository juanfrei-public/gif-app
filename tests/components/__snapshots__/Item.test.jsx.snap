// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre Item Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Un titulo mock"
      src="https://unmock.zaraza.com/"
    />
    <p>
      Un titulo mock
    </p>
  </div>
</div>
`;
